lib/ansible/plugins/inventory/ovirt.py

```diff
@@ -0,0 +1,275 @@
+            :param param_dict: dictionary of filter parameters and values
+            :return dictionary with casted parameter/value
+        '''
+        if param_dict is None:
+            return None
+
+        FILTER_MAPPING = {
+            'all_content': bool,
+            'case_sensitive': bool,
+            'filter': bool,
+            'follow': str,
+            'max': int,
+            'search': str
+        }
+
+        casted_dict = {}
+
+        for (param, value) in param_dict.items():
+            try:
+                casted_dict[param] = FILTER_MAPPING[param](value)
+            except KeyError:
+                raise AnsibleError("Unknown filter option '{0}'".format(param))
+
+        return casted_dict
+
+    def _get_hostname(self, host):
+        '''
+          Get the host's hostname based on prefered attribute
+          :param host: dict representation of oVirt VmStruct
+          :param return: preferred hostname for the host
+        '''
+        hostname_preference = self.get_option('ovirt_hostname_preference')
+        if not hostname_preference:
+            raise AnsibleParserError('Invalid value for option ovirt_hostname_preference: {0}'.format(hostname_preference))
+        hostname = None
+
+        for preference in hostname_preference:
+            hostname = host.get(preference)
+            if hostname is not None:
+                return hostname
+
+        raise AnsibleParserError("No valid name found for host id={0}".format(host.get('id')))
+
+    def _populate_from_source(self, source_data):
+
+        for host in source_data:
+
+            hostname = self._get_hostname(host)
+
+            self.inventory.add_host(hostname)
+
+            for fact, value in host.items():
+                self.inventory.set_variable(hostname, fact, value)
+
+            self._set_composite_vars(self.get_option('compose'), host, hostname)
+            self._add_host_to_composed_groups(self.get_option('groups'), host, hostname)
+            self._add_host_to_keyed_groups(self.get_option('keyed_groups'), host, hostname)
+
+    def verify_file(self, path):
+
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('ovirt.yml', 'ovirt4.yml', 'ovirt.yaml', 'ovirt4.yaml')):
+                valid = True
+        return valid
+
+    def parse(self, inventory, loader, path, cache=True):
+
+        if not HAS_OVIRT_LIB:
+            raise AnsibleError('oVirt inventory script requires ovirt-engine-sdk-python >= 4.2.4')
+
+        super(InventoryModule, self).parse(inventory, loader, path, cache)
+
+        config = self._read_config_data(path)
+
+        self.ovirt_engine_url = self.get_option('ovirt_url')
+        self.ovirt_engine_user = self.get_option('ovirt_username')
+        self.ovirt_engine_password = self.get_option('ovirt_password')
+        self.ovirt_engine_cafile = self.get_option('ovirt_cafile')
```

Comment:
Maybe create here the `self.connection` instead of in the `_get_connection` method.